File: CodeForces/501-600/579/A/579A.go
// Problem 579A: Raising Bacteria
// https://codeforces.com/contest/579/problem/A
// Executed with: go version go1.19.8 linux/amd64
// Linux version: Linux kernel 6.1.0-10-amd64
// Submitted on: October 22nd, 2025
// Codeforces language used: Go 1.22.2

package main

import (
	"fmt"
)

func main() {
	var w int
	fmt.Scan(&w)
	var ans int

	for {
		if w == 0 {
			fmt.Println(ans)
			return
		}

		if w%2 == 0 {
			w /= 2
		} else {
			w -= 1
			w /= 2
			ans += 1
		}
	}
}

File: CodeForces/501-600/579/A/579A.rb
# Problem 579A: Raising Bacteria
# https://codeforces.com/contest/579/problem/A
# Executed with: ruby 3.3.8 (2025-04-09 revision b200bad6cd) [aarch64-linux-gnu]
# Linux version: Linux kernel 6.17.0-5-generic
# Submitted on: October 26th, 2025
# Codeforces language used: Ruby 3.2.2

goal = gets.to_i
ans = 0

while true
  if goal == 0
    break
  end
  if goal.even?
    goal /= 2
  else
    goal -= 1
    ans += 1
  end
end

puts ans
